packages/atomic/src/utils/result-utils.ts

```diff
@@ -42,3 +43,36 @@ export function bindLogDocumentOpenOnResult(
     });
   };
 }
+
+export function buildStringTemplateFromResult(
+  template: string,
+  result: Result,
+  bindings: Bindings
+) {
+  return template.replace(/\${(.*?)}/g, (value: string) => {
+    const key = value.substring(2, value.length - 1);
+    let newValue = readFromObject(result, key);
+    if (!newValue) {
+      newValue = readFromObject(window, key);
+    }
+
+    if (!newValue) {
+      bindings.engine.logger.warn(
+        `${key} used in the href template is undefined for this result: ${result.title}`
+      );
+      return '';
```

Comment:
Except this early return with an empty string, this is taken entirely from the JSUI. Investigated other modern solutions and was getting stuff similar to that or eval,Function stuff which is way riskier